sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs

```diff
@@ -56,13 +57,12 @@ public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePol
             }
         }
 
+        // This method writes HttpMessages to the DiagnosticSource held within DiagnosticScope. This attribute
+        // preserves the public properties on this type so it can be used by ApplicationInsights.
+        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(HttpMessage))]
```

Comment:
Somewhat cleaner pattern is used in other similar places, for example: https://github.com/dotnet/runtime/blob/f9246538e3d49b90b0e9128d7b1defef57cd6911/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L323-L331

The idea is that it would look like (simplified):
```C#
        // Remove the DynamicDependency completely

        using var scope CreateDiagnosticScope(message); // The compiler will infer the generic parameter to be HttpMessage for you - you don't have to type it out.

        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "The values being passed into Write have the commonly used properties being preserved with DynamicDependency.")]
        private DiagnosticScope CreateDiagnosticScope<[DynamicallyAccessedMembers(PublicProperties)] T>(T sourceArgs)
        {
            return new DiagnosticScope("Azure.Core.Http.Request", s_diagnosticSource, sourceArgs, s_activitySource, System.Diagnostics.ActivityKind.Client, false);
        }
```

In general we try to avoid using `DynamicDependency` as it's not verifiable by the tooling - in this case even the DynamicallyAccessedMembers is not verifiable, but it conveys the intent better this way. Also the new version will not break even we start writing a different type into the payload, unlike the old one where the DynamicDependency has to be kept in sync with the rest of the code.

Reply:
Updated!